invenio_users_resources/records/mappings/os-v1/users/user-v3.0.0.json

```diff
@@ -0,0 +1,206 @@
+{
+  "settings": {
+    "analysis": {
+      "analyzer": {
+        "default": {
+          "type": "custom",
+          "tokenizer": "uax_url_email",
+          "filter": [
+            "lowercase"
+          ]
+        },
+        "default_search": {
+          "type": "custom",
+          "tokenizer": "uax_url_email",
+          "filter": [
+            "lowercase"
+          ]
+        },
+        "edge_analyzer": {
+          "tokenizer": "uax_url_email",
+          "filter": [
+            "lowercase",
+            "edgegrams"
+          ]
+        }
+      },
+      "filter": {
+        "edgegrams": {
+          "type": "edge_ngram",
+          "min_gram": 2,
+          "max_gram": 20,
+          "preserve_original": true
+        }
+      }
+    }
+  },
+  "mappings": {
+    "dynamic": "strict",
+    "dynamic_templates": [
+      {
+        "profile": {
+          "path_match": "profile.*",
+          "mapping": {
+            "type": "keyword"
+          }
+        }
+      },
+      {
+        "preferences": {
+          "path_match": "preferences.*",
+          "mapping": {
+            "type": "keyword"
+          }
+        }
+      },
+      {
+        "identities": {
+          "path_match": "identities.*",
+          "mapping": {
+            "type": "keyword"
+          }
+        }
+      }
+    ],
+    "properties": {
+      "$schema": {
+        "type": "keyword",
+        "index": "false"
+      },
+      "id": {
+        "type": "keyword"
+      },
+      "version_id": {
+        "type": "integer"
+      },
+      "uuid": {
+        "type": "keyword"
+      },
+      "created": {
+        "type": "date"
+      },
+      "updated": {
+        "type": "date"
+      },
+      "current_login_at": {
+        "type": "date"
+      },
+      "active": {
+        "type": "boolean"
+      },
+      "confirmed_at": {
+        "type": "date"
+      },
+      "indexed_at": {
+        "type": "date"
+      },
+      "confirmed": {
+        "type": "boolean"
+      },
+      "blocked_at": {
+        "type": "date"
+      },
+      "blocked": {
+        "type": "boolean"
+      },
+      "verified_at": {
+        "type": "date"
+      },
+      "verified": {
+        "type": "boolean"
+      },
+      "username": {
+        "type": "text",
+        "analyzer": "edge_analyzer",
+        "fields": {
+          "keyword": {
+            "type": "keyword"
+          }
+        }
+      },
+      "email": {
+        "type": "text",
+        "analyzer": "edge_analyzer",
+        "fields": {
```

Comment:
good point, I think we should if it helps with the UX

Reply:
To use the `?suggest=` API for users, we will have to update the tests to not follow lucene query syntax since multi_match doesn't support that. This would also mean that lucene query won't work on users anymore.

Reply:
This is fine for the `/api/users` endpoint, which will always be used with `?suggest=...`, which doesn't need Lucene query syntax.

The `/api/users/all` endpoint, which is for the administration backoffice UI, should always use `?q=...` and support Lucene search queries syntax.